internal/service/transactions.go

```diff
@@ -19,91 +19,177 @@ package service
 import (
 	"context"
 	"fmt"
+	"math/big"
```

Comment:
You are eating the return values (error) when setting big.Int from string.

https://pkg.go.dev/math/big@go1.20.3#Int.SetString

I would add a util function that returns only an error and examines the other value